### PR TITLE
Document complex sighting, indicator structures

### PR DESCRIPTION
**Pull request by saintx**
CTIM can be used to model arbitrarily complex nested sightings and
indicators, each of which can be based on other sightings and indicators,
respectively. This tutorial explains and illustrates how to model event
data of arbitrary complexity using sighting and indicator entities.

This work advances the https://github.com/advthreat/hydrant/issues/625 epic.

### CODE REVIEW

File: doc/tutorials/modeling-complex-events-ctim.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Whenever an event on our network or in one of our monitored systems triggers a detection or pattern matching rule, we generate a `sighting` for it. This is true whether your rule is:
+ - a YARA rule in a malware engine
+ - a snort rule running in a network intrusion detection system
+ - a firewall rule
+ - an antivirus or any other kind of filesystem or registry scan on a host endpoint
+ - a monitor for an event log
+ - a rule that's only triggered by *all of the above*.
+
+Ultimately, `sighting` enties are how we record that we saw *anything* trigger any kind of detection or pattern matching rule described by one of our `indicator` entities, at a specific point in time.
```
Comment (frenchy64) [inline doc/tutorials/modeling-complex-events-ctim.md:71]: `enties` typo

File: doc/tutorials/modeling-complex-events-ctim.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+- This summary overview, in the case of a complex pattern match, could also briefly reference the simpler patterns that together triggered the pattern match, but it's not required.
+
+The lesson to learn here is that regardless of how complex your pattern matching or detection rules are, and regardless of whether any one of your rules is composed from a collection of other, similar rules, you can model the information about each pattern matching rule as `indicator` entities in CTIM. Complex rules--where one rule is based on a composition of one or more other rules--are modeled using the `based-on` relationship type.
```
Comment (frenchy64) [inline doc/tutorials/modeling-complex-events-ctim.md:59]: `other, similar rules` reads a bit awkwardly to me, maybe drop the comma and/or parenthesize `similar`.

File: doc/tutorials/modeling-complex-events-ctim.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+CTIM is capable of modeling arbitrarily complex event data. That is, events that are triggered by other events, which themselves might be triggered by other events.
+
+## Synopsis
+
+- Arbitrarily complex cybersecurity events can be modeled in CTIM.
+- Pattern matching or detection rules are modeled with the `indicator` entity.
+- An `indicator` entity can be `based-on` other `indicator` entities.
+- When modeling `indicator` entities, less is more.
+  - Don't include the patterns you match against in your `indicator` entities. They'll be used for detection evasion.
+  - The `indicator` entity should only summarize the threat. Using other entities for describing threats in depth.
```
Comment (frenchy64) [inline doc/tutorials/modeling-complex-events-ctim.md:12]: `Using`->`--use`

File: doc/tutorials/modeling-complex-events-ctim.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+- When a pattern matching or detection rule described by an `indicator` is triggered, we create a `sighting` that is a `sighting-of` relationship to that `indicator`.
+- A `sighting` entity can be `based-on` other `sighting` entities, in addition to being a `sighting-of` an `indicator`.
+
+## Indicators as Detection or Pattern Matching Rules
+
+There are two basic indicator types. This tutorial concerns itself with the first type, which is a detection or pattern matching rule. These rules look for patterns in data, and can run inside of rule engines, expert systems, firewalls, log monitors, or any other kind of scanning or monitoring system.
```
Comment (frenchy64) [inline doc/tutorials/modeling-complex-events-ctim.md:18]: `There are two basic indicator types.` -> `There are two basic indicator types: either a detection or pattern matching rule, or a ??`.

File: doc/tutorials/modeling-complex-events-ctim.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+Pattern matching rules are often modeled using composition, such that a series of pattern matches within a given timeframe can combine to trigger more complex pattern matches. Here are three contrived examples:
```
Comment (frenchy64) [inline doc/tutorials/modeling-complex-events-ctim.md:20]: Perhaps there's a better word than `composition` here. Aggregation? I had trouble relating the two halves of the sentence.

**Review by frenchy64 — APPROVED** [review]
Very nice and clear, I learnt something. Only pedantry for feedback.

File: doc/tutorials/modeling-complex-events-ctim.md (hunk 4 above)
Comment (saintx) [reply doc/tutorials/modeling-complex-events-ctim.md:18]: I'll link to other docs on this one. I don't want to discuss the other type (the observable watch list) in this tutorial.

File: doc/tutorials/modeling-complex-events-ctim.md (hunk 5 above)
Comment (saintx) [reply doc/tutorials/modeling-complex-events-ctim.md:20]: Too many verbs. Tightened the prose.

**Pull request merged**